firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -205,13 +205,16 @@ private void writeFIDCreateRequestBodyToOutputStream(
 
   private static void writeRequestBodyToOutputStream(URLConnection urlConnection, byte[] jsonBytes)
       throws IOException {
-    try (OutputStream outputStream = urlConnection.getOutputStream()) {
-      if (outputStream == null) {
-        throw new IOException("Cannot send request to FIS servers. No OutputStream available.");
-      }
-      try (GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream)) {
-        gzipOutputStream.write(jsonBytes);
-      }
+    OutputStream outputStream = urlConnection.getOutputStream();
+    if (outputStream == null) {
+      throw new IOException("Cannot send request to FIS servers. No OutputStream available.");
+    }
+    GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream);
+    try {
+      gzipOutputStream.write(jsonBytes);
+    } finally {
+      gzipOutputStream.close();
+      outputStream.close();
```

Comment:
2 Questions:
a) Do we want to catch the IOExceptions here?
Thought process is this:
If an IOException is thrown during closing, do we really want to interrupt the normal process of FIS?
b) If your answer to (a) is "no", could you please add a comment explaining your decision?

Reply:
Caught the exception. But, I hate these checks and code. It was so much cleaner and easy to maintain before.